ras_backstage/controllers/collection_exercise_controller.py

```diff
@@ -41,3 +41,32 @@ def get_collection_exercises_by_survey(survey_id):
 
     logger.debug('Successfully retrieved collection exercises', survey_id=survey_id)
     return json.loads(response.text)
+
+
+def get_linked_sample_summary_id(collection_exercise_id):
+    logger.debug('Retrieving sample linked to collection exercise', collection_exercise_id=collection_exercise_id)
+    url = f'{app.config["RM_COLLECTION_EXERCISE_SERVICE"]}collectionexercises/link/{collection_exercise_id}'
+    response = request_handler('GET', url, auth=app.config['BASIC_AUTH'])
+
+    if response.status_code == 404:
+        logger.error('Error retrieving collection exercise', collection_exercise_id=collection_exercise_id)
+        raise ApiError(url, response.status_code)
+    elif response.status_code == 204:
+        logger.error('No samples linked to collection exercise', collection_exercise_id=collection_exercise_id)
+        raise ApiError(url, response.status_code)
+    elif response.status_code != 200:
+        logger.error('Error retrieving sample summaries linked to collection exercise',
+                     collection_exercise_id=collection_exercise_id)
+        raise ApiError(url, response.status_code)
+
+    try:
+        sample_summary_id = response.json()[0]
```

Comment:
Do we need the try-except here. If there's no samples linked shouldn't that be caught with the 204 status code?

Reply:
@andrewmil it *should* be but I'd rather catch a sudden change to the API and return a 404 than cause an unexpected exception to bubble up. Unless that's preferred?